sklearn/gaussian_process/kernels.py

```diff
@@ -1852,7 +1852,7 @@ def diag(self, X):
             Diagonal of kernel k(X, X)
         """
         # We have to fall back to slow way of computing diagonal
-        return np.apply_along_axis(self, 1, X)[:, 0]
+        return np.apply_along_axis(lambda arr: self(arr).ravel(), 1, X)[:, 0]
```

Comment:
I think this is written in a confusing way, if `self(arr)` is returning an array of shape (1,1).

Shouldn't either of the following do the same?
```python
np.apply_along_axis(lambda arr: self(arr)[0, 0], 1, X)
np.apply_along_axis(self, 1, X).ravel()
```